### ISSUE TITLE
Session Restore Problems

### ISSUE THREAD

**Issue by cenk1cenk2** [kyazdani42/nvim-tree.lua#296]
Hello,

Initially, thank you for your continuous work on this beautiful approach to file explorer.

I have recently started to encounter problems with session management after the commit sha `bbb8d60`. When I load it with `nvim .` the nvim-tree works fine but when I save that session and reload it, it throws out a buffer id unknown error (not always) and just toggles a split of the current buffer. I have searched through the issues but this buffer id unknown seems to be a rather old issue [here](https://github.com/kyazdani42/nvim-tree.lua/issues/175) which might be unrelated.

**Comment by kyazdani42** [comment]
Hi, could you provide me with some steps to reproduce this please ? i modified the tree view and buffer handling which might cause some instabilities (also known as bugs) until this is completely stable for everyone.

**Comment by cenk1cenk2** [comment]
Hello,

I tried to reproduce it with today's last commit `f7047d6`, and tried it with `mksession` and can not reproduce it.

But while using other plugins to manage sessions like `glepnir/dashboard-nvim`, `mhinz/vim-startify` or `rmagatti/auto-session` the issue persists. I am not sure what is actually wrong with it and how to reproduce it on vanilla configuration, since these plugins session management should be a wrapper around `mksession`.

The end result looks something like this when `nvim-tree` is triggered.
![image](https://user-images.githubusercontent.com/26881592/114888797-d75ca780-9e09-11eb-926d-88ddd5f77713.png)

**Comment by kyazdani42** [comment]
i've noticed there is a startup issue with nvim-tree, not sure why, i'm going to do some debugging, thanks for the informations !

**Comment by kyazdani42** [comment]
couldn't reproduce your issue even on the commit you specify with either glepnir / rmagatti plugins. Not sure why this behaves like this. Could you provide a minimal config with which you have this issue popping up ?

**Comment by cenk1cenk2** [comment]
I can provide the full config that I currently use [here](https://github.com/cenk1cenk2/nvim), since you are currently having a look into it. I will try to reproduce it with a more minimal setup this weekend. Thank you again for looking into this.

**Comment by kyazdani42** [comment]
i've looked at you config, but don't really have the time to dig deep into it :) let me know when you have a minimal setup reproducible

**Comment by akinsho** [comment]
I just hit this issue as well, so can confirm it's definitely happening. It's quite late here so will try and pass on a report later. In terms of my setup in brief I'm using `dashboard` and it's session management logic when I start a session that way. It also doesn't reproduce without a session. I'll try and get a minimal version when I have time tomorrow

**Comment by kyazdani42** [comment]
thanks akinsho, this issue might come from the speed of nvim-tree which might have gone up again, because the buffer is created almost at first on startup, so maybe deferring the initalization of the plugin to later might fix it ?

**Comment by akinsho** [comment]
Hmm not really sure I just managed to reproduce it by removing every other plugin and just leaving, `nvim-tree`, `dashboard.nvim`. You can then open some files, use `SessionSave` then close nvim, the next time use `SessionLoad` and then toggle the tree and I get

```
E5108: Error executing lua ...e/pack/packer/start/nvim-tree.lua/lua/nvim-tree/view.lua:142: Vim(buffer):E86: Buffer 2 does not exist
```
If i check `ls!` the list is
```
:ls!
  4 %a   ".config/nvim/lua/as/plugins/init.lua" line 390
  5      ".config/nvim/lua/as/plugins/compe.lua" line 85
  6      ".config/nvim/lua/as/plugins/gitsigns.lua" line 11
  7      ".config/nvim/lua/as/autocommands.lua" line 3
  8      ".config/nvim/lua/as/globals/utils.lua" line 238
  9      ".config/nvim/lua/as/plugins/nvim-tree.lua" line 16
 10      "install.conf.yaml"            line 28
 11      "~/.dotfiles/.config/nvim/lua/as/plugins/telescope.lua" line 175
 12  h   "[No Name]"                    line 0
 13  h   "[No Name]"                    line 0
```
It looks like the buffer got wiped out somehow maybe when the session got loaded. I think vim clears everything then loads the buffers from the session file and so the buffer allocated on startup for nvim tree and saved is no longer there. Might be a good idea to have some autocommand hook so if the `View.bufnr` is not valid assign a new one? anyway gotta sleep now will dig back in later

**Comment by kyazdani42** [comment]
i guess sindrets might have fixed that already, could you guys try his PR ?

**Comment by edementyev** [comment]
> i guess sindrets might have fixed that already, could you guys try his PR ?

Just checked out his PR, it works great! Was having that issue and now it's gone.
Thank you for your work on this beautiful plugin! Enjoying it very much.

**Comment by cenk1cenk2** [comment]
I can confirm that this is gone for me thank you very much @sindrets , @kyazdani42.

**Comment by WIttyJudge** [comment]
I fixed this problem by adding this to my lua config.
```
vim.o.sessionoptions="blank,buffers,curdir,folds,help,options,tabpages,winsize,resize,winpos,terminal"
```

If you use vim, try to add this instead of example above:
```
set sessionoptions+=options,resize,winpos,terminal
```

**Comment by dvogt23** [comment]
Have the same issue with the newest version:
```
  3 %a   "[No Name]"                    line 1
  4  a   "[No Name]"                    line 0
  7u     "NvimTree"                     line 1
startuptime:  0.424192
E5108: Error executing lua ...e/pack/packer/start/nvim-tree.lua/lua/nvim-tree/view.lua:306: Vim(buffer):E86: Buffer 2 does not exist
```

**Comment by kyazdani42** [comment]
@dvogt23 should be fixed in 1.1.5 (4cc2abe)
